### DISCUSSION TITLE
Relation Manager client > order > order_line

### DISCUSSION THREAD

**Discussion by micraux** [filamentphp/filament#8416] [Help] [answered]
I made some researches inside the discussions and it seems some people achieved what I am trying to do, but I'm stuck...

I've got a `Client` resource with a Relation Manager for the `Orders`, and I would like to open a modal when I click an Order, containing the Order details (date, invoice number...) and the Order Lines.

I've created a custom page and a Livewire Component, and am trying to call the custom page in a modal with an Action on the Order Id cell.

But when I try to pass the Order record I get an error: it looks like I am calling the `App\Models\Order` instead of the `ClientResource\RelationManagers\Order`

      ->action(
          Action::make('details_order')
              ->action(function (Order $record): void {
                  $this->dispatch('details-order', order: $record->getKey());
              })
      ),

Any clue?

**Comment by leandrocfe** [comment]
Would you like to do this?

https://demo.filamentphp.com/blog/posts/2/edit

![image](https://github.com/filamentphp/filament/assets/3833889/94b04685-8ce5-4d2e-9a40-c9e039a0bf92)
![image](https://github.com/filamentphp/filament/assets/3833889/73acee90-6e54-4cea-90ca-6ff63109cf5b)

**Comment by micraux** [comment]
Yes, with just another level of information: in the demo, you have `Posts` > `Comments`.
I would like to have `Clients` > `Orders` but with a `Order Lines table` inside the modal.

But for now, I cannot even open the modal with just the `Order` details :-(
I know I am missing something but can't figure how to achieve that.

**Comment by micraux** [comment]
What I need would be basically this page in the demo: https://demo.filamentphp.com/shop/orders

But I would like, when you click an Order, is a modal showing (not editing) the order details and the order products (simple table, not repeater)...

**Reply by leandrocfe** [reply]
*when you click an Order, is a modal showing (not editing) the order details*
- Does it have to be a modal, or can it be a page, like an 'EditPage' with disabled fields?

*the order products (simple table, not repeater)*
- If the order is a page, you can define a 'RelationManager' for products, similar to the 'Payment' example in this case.

**Comment by micraux** [comment]
Hello @leandrocfe,
1. I know I can use a `View` page but would prefer a modal, just for the user to see the details (I will later create a separate edit page)
2. Yes, I know about the `Relation Manager`, but that will not solve my problem as I need to show the `Products` (or `Order Lines`) in a table inside the modal

Basically, I need the same thing as in discussion https://github.com/filamentphp/filament/discussions/8277. That is why I followed your advice and created a custom page and a livewire component.

Because, if I understood correctly, you can't put a table inside a modal, so you have to create a custom page for the modal.
Or am I wrong?

By the way, I am going to see if I can download the demo to see the code: I'm pretty sure that will help me!

**Reply by leandrocfe** [reply]
Take a look: https://github.com/filamentphp/filament/discussions/8293#discussioncomment-6929362

**Comment by micraux** [comment]
I'll have a look, thanks!

**Reply by micraux** [reply]
Not working... I guess it is because ti has been written for v2 and there are some differences...
I will try again from scratch, following the documentation step by step...

**Reply by leandrocfe** [reply]
This was supposed to work on v2, but I agree with *scratch, following the documentation step by step*
If you want, share your repo on Github

**Reply by micraux** [reply]
I won't be able to do it, because in fact this is not about Orders and Order Lines but my English is not good enough to explain it :-)
It is a system where the company leaves books in a point of sell, comes back the week after and counts how many books are missing (sold by the point of sale) and then make an invoice based on the sold books.
But the DB structure (in my case Client > Dépôt > Livres) is the same as Client > Order > Product

**Comment by micraux** [comment]
**1. Create an action for the `Orders` table (created by the Client Relation Manager) to open a modal with a custom view**

I follow the example in https://filamentphp.com/docs/3.x/actions/modals#passing-data-to-the-custom-modal-content but created a simple view instead of a page, so I did:
```
->actions([
    Action::make('détails')
        ->action(fn (Order $record) => $record->advance())
        ->modalContent(fn (Order $record): View => view(
            'filament.views.order-details',
            ['record' => $record],
        ))
])
```
And in my view:
```
<div>
    <p>{{ $record['id'] }}</p>
</div>
```
This is working: I can see my Order Id in the modal.

For the moment, is it the "good" way to start?

**Reply by leandrocfe** [reply]
I've never tried this before, but it is possible I think
You can try to use the livewire component inside the view to render the table

**Reply by micraux** [reply]
Well, it is working, and of course creating the LW component will be my second step :-)

**Reply by micraux** [reply]
I asked if it was a good start because I don't know if I should pass the Model or the Resource in this step...

**Comment by micraux** [comment, marked as answer]
**2. Create the LW component and add a table to show the Order Lines**
I follow the example in https://filamentphp.com/docs/3.x/tables/adding-a-table-to-a-livewire-component

I managed to make it work but there might be an error in the `render()` method because I got an error saying that the view doesn't exist and I have to add the LW directory to make it work, like: `return view('livewire.list-products');`. Do you think that makes sense?

I just can't figure how to retrieve the $record in my LW component. I passed my record like this:
 `@livewire('depot-details', $record->toArray())`. Is it the good way to do it?

**Reply by micraux** [reply]
Ok, I managed to pass the $record using `@livewire('depot-details', ['record' => $record])`

And I can retrieve it in the `mount()` method inside the LW component

**Reply by micraux** [reply]
Well, it works!

Thank you very much @leandrocfe for your time and patience!

@danharrin: if you have some time, would you tell me if I made it correctly, or if there is a better approach?

Since I struggled on the "livewire" segment in the render() method and the way to retrieve the $record in the mount() methods, I thought I may share the process in the "Show and tell" section, if you think that could help...

**Reply by leandrocfe** [reply]
As I mentioned earlier, my preference is to use tables within a page. This approach is ok to me and it's just another way to achieve the result.
